### PR TITLE
Fix: vela CLI provider compatibility

### LINKED ISSUE #4558: CLI provider command failed
**Describe the bug**

There are incompatibility for Vela CLI v1.5.0 and terraform-xxx in newer version. After enable terraform-alibaba, provider type componentDefinition can't be found.

```shell
$ vela provider list
Error: no Terraform Cloud Provider found, please run `vela addon enable` first
```

**Reason**

Removal of `custom.definition.oam.dev` prefix effect the `vela provider` command. It uses labels to select components.

**KubeVela Version**

Vela CLI 1.5.0
terraform-alibaba Addon: v1.0.3

**Cluster information**

VelaD v1.5.0

**Additional context**

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: qiaozp <qiaozhongpei.qzp@alibaba-inc.com>

### Description of your changes

In vela CLI < v1.5.0, `vela addon enable` will add a prefix ("custom.definition.oam.dev") to label of definition came with addon.  This logic is removed in v1.5.0

However it effect `vela provider list` behavior. Which still reading ComponentDefinition with Label `"custom.definition.oam.dev:type.config.oam.dev": terraform-xxx`. This PR make provider command compatible with both type of label.

Fixes #4558

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

UT added

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
good job

File: references/cli/provider.go [hunk 1]
```diff
@@ -297,12 +297,27 @@ func listProviders(ctx context.Context, k8sClient client.Client, ioStreams cmdut
 // getTerraformProviderTypes retrieves all ComponentDefinition for Terraform Cloud Providers which are delivered by
 // Terraform Cloud provider addons
 func getTerraformProviderTypes(ctx context.Context, k8sClient client.Client) ([]v1beta1.ComponentDefinition, error) {
+	// keep compatibility with old version of terraform-xxx provider definition
+	legacyDefs := &v1beta1.ComponentDefinitionList{}
+	err := k8sClient.List(ctx, legacyDefs, client.InNamespace(types.DefaultKubeVelaNS),
+		client.MatchingLabels{definition.UserPrefix + definition.DefinitionType: types.TerraformProvider})
+	if err != nil && kerrors.IsNotFound(err) {
```
Comment (charlie0129) [inline references/cli/provider.go:304]:
```suggestion
	if err != nil && !kerrors.IsNotFound(err) {
```

File: references/cli/provider.go [hunk 2]
```diff
@@ -297,12 +297,27 @@ func listProviders(ctx context.Context, k8sClient client.Client, ioStreams cmdut
...
+		return nil, err
+	}
 	defs := &v1beta1.ComponentDefinitionList{}
-	if err := k8sClient.List(ctx, defs, client.InNamespace(types.DefaultKubeVelaNS),
-		client.MatchingLabels{definition.UserPrefix + "type.config.oam.dev": types.TerraformProvider}); err != nil {
+	err = k8sClient.List(ctx, defs, client.InNamespace(types.DefaultKubeVelaNS),
+		client.MatchingLabels{definition.DefinitionType: types.TerraformProvider})
+	if err != nil {
```
Comment (charlie0129) [inline references/cli/provider.go:310]:
```suggestion
	if err != nil && !kerrors.IsNotFound(err) {
```

Comment (chivalryq) [reply references/cli/provider.go:310]: nice catch! God.

**Review by charlie0129 — APPROVED** [review]
Great fix! LGTM

File: references/cli/provider_test.go [hunk 3]
```diff
@@ -21,6 +21,10 @@ import (
 	"os"
 	"testing"
 
+	"github.com/pkg/errors"
+
+	"github.com/oam-dev/kubevela/pkg/definition"
```
Comment (charlie0129) [inline references/cli/provider_test.go:26]: unformatted imports here

File: references/cli/provider.go (hunk 2 above)
Comment (chivalryq) [reply references/cli/provider.go]: List Interface won't return `NotFoundError` 😂

Comment (charlie0129) [reply references/cli/provider.go]: oh good to know

**Review by FogDong — APPROVED** [review]

**Pull request merged**